pulseapi/users/views.py

```diff
@@ -63,16 +63,41 @@ def new_nonce_value(request):
 def nonce(request):
     """
     set a new random nonce to act as form post identifier
+    and inform the user what this value is so they can use
+    it for signing their POST for a new entry.
     """
     if not request.user.is_authenticated():
-        return HttpResponseNotFound()
+        return HttpResponse('Not authorized', status=403)
 
     new_nonce_value(request)
-    return render(request, 'users/formprotection.json', {
-        'user': request.user,
+    return render(request, 'users/nonce.json', {
         'nonce': request.session['nonce']
     }, content_type="application/json")
 
+
+# API ROUTE: /userstatus
+def userstatus(request):
+    """
+    Get the login status associated with a session. If the
+    status is "logged in", also include the user name and
+    user email. NOTE: these values should never be persistently
+    cached by applications, for obvious reasons.
+    """
+    name = False
+    email = False
+    loggedin = request.user.is_authenticated()
+
+    if loggedin:
+        name = request.user.name
+        email = request.user.email
+
+    return render(request, 'users/userstatus.json', {
```

Comment:
Django needs a template to render JSON, it can't just render the JSON straight up?